src/components/FilterByCentury/CenturyButton.tsx

```diff
@@ -0,0 +1,34 @@
+import classNames from 'classnames';
+import { SearchLink } from '../SearchLink/SearchLink';
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+type Props = {
+  century: string;
+};
+
+export const CenturyButton: React.FC<Props> = ({ century }) => {
+  const [searchParams] = useSearchParams();
+  let centuries = searchParams.getAll('centuries');
```

Comment:
The `centuries` array is being directly modified later in the code. Consider using a new array to maintain immutability, such as by using the spread operator or `Array.from()` to create a copy before modifying it.